src/main/java/com/github/mc1arke/sonarqube/plugin/ce/pullrequest/azuredevops/AzureDevOpsServerPullRequestDecorator.java

```diff
@@ -0,0 +1,414 @@
+                handleIssue(analysisDetails, azurePullRequestDetails, issue, azureCommentThreads);
+            }
+            return DecorationResult.builder().withPullRequestUrl(getPullRequestUrl(azurePullRequestDetails, false)).build();
+        } catch (Exception ex) {
+            throw new IllegalStateException("Could not decorate Pull Request on AzureDevOps Server", ex);
+        }
+    }
+
+    private void handleIssue(AnalysisDetails analysisDetails, AzurePullRequestDetails azurePullRequestDetails,
+                             PostAnalysisIssueVisitor.ComponentIssue issue, ArrayList<CommentThread> azureCommentThreads) {
+        String filePath = analysisDetails.getSCMPathForIssue(issue).orElse(null);   
+
+        if (filePath != null) {
+            try {
+                Integer line = issue.getIssue().getLine();
+                filePath = "/" + filePath;
+                LOGGER.trace(String.format("ISSUE: authorLogin: %s ", issue.getIssue().authorLogin()));
+                LOGGER.trace(String.format("ISSUE: key: %s ", issue.getIssue().key()));
+                LOGGER.trace(String.format("ISSUE: type: %s ", issue.getIssue().type().toString()));
+                LOGGER.trace(String.format("ISSUE: severity: %s ", issue.getIssue().severity()));
+                LOGGER.trace(String.format("ISSUE: componentKey: %s ", issue.getIssue().componentKey()));
+                LOGGER.trace(String.format("ISSUE: getLocations: %s ", Objects.requireNonNull(issue.getIssue().getLocations()).toString()));
+                LOGGER.trace(String.format("ISSUE: getRuleKey: %s ", issue.getIssue().getRuleKey()));
+                LOGGER.trace(String.format("COMPONENT: getDescription: %s ", issue.getComponent().getDescription()));
+                DbIssues.Locations locate = Objects.requireNonNull(issue.getIssue().getLocations());
+                boolean threadExists = false;
+                for (CommentThread azureThread : azureCommentThreads) {
+                    LOGGER.trace(String.format("azureFilePath: %s", azureThread.getThreadContext().getFilePath()));
+                    LOGGER.trace(String.format("filePath: %s (%s)", filePath, azureThread.getThreadContext().getFilePath().equals(filePath)));
+                    LOGGER.trace(String.format("azureLine: %d", azureThread.getThreadContext().getRightFileStart().getLine()));
+                    LOGGER.trace(String.format("line: %d (%s)", line, azureThread.getThreadContext().getRightFileStart().getLine() == locate.getTextRange().getEndLine()));
+
+                    // Check if thread already exists and close thread if issue is closed
+                    if (checkAzureThread(issue, filePath, azureThread, azurePullRequestDetails)) {
+                        threadExists = true;
+                        break;
+                    }
+                }
+                if (issue.getIssue().getStatus() != Issue.STATUS_OPEN) {
```

Comment:
I think you should use equals(), because status is not enum but string :(

Reply:
Fixed